src/azureFunction/azureFunctionUtils.ts

```diff
@@ -76,8 +76,48 @@ export async function setLocalAppSetting(projectFolder: string, key: string, val
 	return true;
 }
 
+/**
+ * Gets the Azure Functions extension API if it is installed
+ * if it is not installed, prompt the user to install directly, learn more, or do not install
+ * @returns the Azure Functions extension API if it is installed, prompt if it is not installed
+ */
 export async function getAzureFunctionsExtensionApi(): Promise<AzureFunctionsExtensionApi | undefined> {
-	const apiProvider = await vscode.extensions.getExtension(constants.azureFunctionsExtensionName)?.activate() as AzureExtensionApiProvider;
+	let apiProvider = await vscode.extensions.getExtension(constants.azureFunctionsExtensionName)?.activate() as AzureExtensionApiProvider;
+	if (!apiProvider) {
+		const response = await vscode.window.showInformationMessage(constants.azureFunctionsExtensionNotFound,
+			constants.installAzureFunction, constants.learnMore, constants.doNotInstall);
+		if (response === constants.installAzureFunction) {
+			await vscode.window.withProgress(
+				{
+					location: vscode.ProgressLocation.Notification,
+					title: constants.azureFunctionsExtensionName,
+					cancellable: false
+				}, async (_progress, _token) => {
+					await vscode.commands.executeCommand('workbench.extensions.installExtension', constants.azureFunctionsExtensionName);
+				}
+			);
+			// the extension has not been notified that the extension is installed so wait till it is to then activate it
+			await new Promise<void>((resolve, reject) => {
```

Comment:
You should create (but not await) this promise before the `withProgress` call so that the event is hooked up in case there's a race condition here and we end up having the extension host updated before the withProgress returns. 

(and then await it down here)